test/test_nn.py

```diff
@@ -11392,6 +11390,175 @@ def __init__(self):
+        output_2.backward(torch.ones(5, 5) * 2, retain_graph=True)
+        output_3.backward(torch.ones(5, 5) * 2, retain_graph=True)
+        self.assertEqual(counter['forwards'], 6)
+        self.assertEqual(counter['backwards'], 3)
+
+        output_1.backward(torch.ones(5, 5) * 2, retain_graph=True)
+        self.assertEqual(counter['forwards'], 6)
+        self.assertEqual(counter['backwards'], 4)
+
+        test2_fwd = module.register_global_forward_hook(lambda *args: fw_hook(2, *args))
+
+        output = module_1(input)
+        output = module_2(input)
+        output = module_3(input)
+        self.assertEqual(counter['forwards'], 15)
+        self.assertEqual(counter['backwards'], 4)
+
+        test2_bwd = module.register_global_backward_hook(lambda *args: bw_hook(2, *args))
+
+        module_1(input).backward(torch.ones(5, 5) * 2)
+        self.assertEqual(counter['forwards'], 18)
+        self.assertEqual(counter['backwards'], 7)
+
+        test2_bwd.remove()
+
+        module_2(input).backward(torch.ones(5, 5) * 2)
+        self.assertEqual(counter['forwards'], 21)
+        self.assertEqual(counter['backwards'], 8)
+
+        test2_fwd.remove()
+
+        module_3(input).backward(torch.ones(5, 5) * 2)
+        self.assertEqual(counter['forwards'], 22)
+        self.assertEqual(counter['backwards'], 9)
+
+        test_fwd.remove()
+        test_bwd.remove()
+
+    def test_module_global_hook_cpp(self):
+        self._clear_hooks()
+        counter = [0]
+        bn = nn.BatchNorm1d(5)
+
+        def hook(module, grad_inputs, grad_outputs):
+            counter[0] += 1
+            self.assertEqual(len(grad_inputs), 3)
+            self.assertEqual(len(grad_outputs), 1)
+            self.assertEqual(module, bn)
+
+        nn.BatchNorm1d.register_global_backward_hook(hook)
+        output = bn(torch.randn(5, 5, requires_grad=True))
+        output.sum().backward()
+
+    def test_module_global_hook_fail(self):
+        self._clear_hooks()
+        module = nn.Sigmoid()
+        input = torch.randn(5, 5, requires_grad=True)
+
+        def bw_fail1(self, grad_input, grad_output):
+            return grad_input[:-1]
+
+        def bw_fail2(self, grad_input, grad_output):
+            return grad_input + (torch.randn(2, 2),)
+
+        with nn.Sigmoid.register_global_backward_hook(bw_fail1):
+            with self.assertRaises(RuntimeError) as err:
+                module(input).sum().backward()
+            self.assertIn("bw_fail", err.exception.args[0])
+            self.assertIn("got 0, but expected 1", err.exception.args[0])
+
+        with nn.Sigmoid.register_global_backward_hook(bw_fail2):
+            with self.assertRaises(RuntimeError) as err:
+                module(input).sum().backward()
+            self.assertIn("bw_fail2", err.exception.args[0])
+            self.assertIn("got 2, but expected 1", err.exception.args[0])
+
+    def test_module_global_hook_writeable(self):
```

Comment:
Can you mention in the test name that this one is for backward hook only and the one below for forward and pre_forward only. And use the same name: either writable or mutations. But since they test the same thing, keep the same name.